service/operator/framework.go

```diff
@@ -39,33 +40,45 @@ func newFramework(config Config) (*framework.Framework, error) {
 		}
 	}
 
-	var resourceGroupResource framework.Resource
+	var resourceGroupResource *resourcegroup.Resource
 	{
-		resourceGroupConfig := resourcegroup.DefaultConfig()
-		resourceGroupConfig.AzureConfig = config.AzureConfig
-		resourceGroupConfig.Logger = config.Logger
+		c := resourcegroup.DefaultConfig()
+		c.AzureConfig = config.AzureConfig
+		c.Logger = config.Logger
 
-		resourceGroupResource, err = resourcegroup.New(resourceGroupConfig)
+		resourceGroupResource, err = resourcegroup.New(c)
 		if err != nil {
 			return nil, microerror.Maskf(err, "resourcegroup.New")
 		}
 	}
 
-	var deploymentResource framework.Resource
+	var deploymentResource *deployment.Resource
```

Comment:
Why this change?

Reply:
I believe exposing the original type as long as possible is a good thing, because hiding it behind an interface hides implementation details.

Reply:
Disagree and commit.